src/uopool/services/simulation.rs

```diff
@@ -151,9 +330,23 @@ where
             }
         })?;
 
+        let mut stake_info_by_entity: HashMap<usize, StakeInfo> = HashMap::new();
```

Comment:
I think fixed array `[StakeInfo;3]` is a better data structure here.

Reply:
Is there a good reason to use HashMap here?

Reply:
Yes, you're right. Will change that.